api/v1alpha1/clustertemplate_types.go

```diff
@@ -81,7 +82,7 @@ type ClusterTemplateStatus struct {
 //+kubebuilder:subresource:status
 
 // ClusterTemplate is the Schema for the clustertemplates API
-// +kubebuilder:validation:XValidation:message="Spec changes are not allowed for a ClusterTemplate that has passed the validation", rule="!has(oldSelf.status) || oldSelf.status.conditions.exists(c, c.type=='ClusterTemplateValidated' && c.status=='False') || oldSelf.spec == self.spec"
+// +kubebuilder:validation:XValidation:message="Spec changes are not allowed for a ClusterTemplate that has passed the validation", rule="!has(oldSelf.status) || oldSelf.status.conditions.exists(c, c.type=='ClusterTemplateValidated' && c.status=='False') || (oldSelf.spec.name == self.spec.name && oldSelf.spec.description == self.spec.description && oldSelf.spec.version == self.spec.version && oldSelf.spec.characteristics == self.spec.characteristics && oldSelf.spec.metadata == self.spec.metadata && oldSelf.spec.templates == self.spec.templates && oldSelf.spec.templateParameterSchema == self.spec.templateParameterSchema)"
```

Comment:
Not sure why this needs to change? Is this to allow updating templateUUID even after template has been validated?

Reply:
Without this change, the validation after the uuid is updated (in case it was missing).

Reply:
Fixed. @Missxiaoguo thanks for the retest. I think we can revert this. I belive it was due to templateID being present in my early testing